### PR TITLE
Cover with cypress tests server search display scenarios part 1

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
What it says on the tin. Implements all scenarios listed on the ticket bellow.

#### Ticket Link
implements: https://mattermost.atlassian.net/browse/MM-30385

### CODE REVIEW

File: e2e/cypress/integration/search/post_search_display_spec.js [hunk 1]
```diff
@@ -37,21 +50,223 @@ describe('Post search display', () => {
...
+
+        // # RHS should be visible with search results
+        cy.get('#search-items-container').should('be.visible');
+
+        // * Assert search results are present and correct
+        cy.get('[data-testid="search-item-container"]').should('be.visible').should('contain.text', '#hello');
+    });
+
+    it('MM-T2291 - Wildcard Search', () => {
+        cy.wait(TIMEOUTS.HALF_SEC);
```
Comment (nronas) [inline e2e/cypress/integration/search/post_search_display_spec.js:229]:
```suggestion
```

File: e2e/cypress/integration/search/post_search_display_spec.js [hunk 2]
```diff
@@ -37,21 +50,223 @@ describe('Post search display', () => {
...
+        const testMessage = 'Hello World!!!';
```
Comment (nronas) [inline e2e/cypress/integration/search/post_search_display_spec.js:230]:
```suggestion
        const testMessage = 'Hello World!!!';
```

File: e2e/cypress/integration/search/post_search_display_spec.js [hunk 3]
```diff
@@ -10,24 +10,37 @@
 // Stage: @prod
```
Comment (saturninoabril) [inline e2e/cypress/integration/search/post_search_display_spec.js:10]: Please remove this prod metadata. We'll add it back once we found the tests added here to be stable for production test.
```suggestion
```

File: e2e/cypress/integration/search/post_search_display_spec.js [hunk 4]
```diff
@@ -37,21 +50,222 @@ describe('Post search display', () => {
...
+        cy.get('[data-testid="search-item-container"]').should('be.visible').should('contain.text', '#hello');
+    });
+
+    it('MM-T2291 - Wildcard Search', () => {
+        const testMessage = 'Hello World!!!';
+
+        // # Post message
+        cy.postMessage(testMessage);
+
+        // # Search for search term in:town-square{space}
```
Comment (saturninoabril) [inline e2e/cypress/integration/search/post_search_display_spec.js:234]:
```suggestion
        // # Search for `Hell*`
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @nronas! Looks good to me except for few requests.

File: e2e/cypress/integration/search/post_search_display_spec.js [hunk 5]
```diff
@@ -37,21 +50,222 @@ describe('Post search display', () => {
...
+    it('MM-T350, MM-T2283 - Searching displays results in RHS', () => {
+        enableTestCommands(testTeam);
+
+        const testSearch = '/test url test-search';
+
+        // # Post a message
+        cy.postMessage(testSearch);
+
+        // # Search the world 'hello' that is generated by /test command
+        cy.get('#searchBox').type('hello').type('{enter}').should('have.value', 'hello');
```
[the remark this answers is not in the document]
Comment (nronas) [reply e2e/cypress/integration/search/post_search_display_spec.js:70]: Will do, thanks a lot for looking into it.

**Review by saturninoabril — APPROVED** [review]
Thanks @nronas! Tested and all passed. LGTM.

**Pull request merged**